torch/_inductor/ir.py

```diff
@@ -2666,12 +2734,25 @@ def codegen(self, wrapper):
             self.cpp_kernel,
         )
 
-    def __init__(self, layout, inputs, constant_args=(), kwargs=None, output_view=None):
+    def __init__(
+        self,
+        layout,
+        inputs,
+        constant_args=(),
+        kwargs=None,
+        output_view=None,
+        kernel=None,
+        cpp_kernel=None,
+    ):
         super().__init__(
             None, layout, self.unwrap_storage(inputs), constant_args, kwargs or {}
         )
         self.output_view = output_view
         self.name = V.graph.register_buffer(self)
+        if kernel is not None:
+            self.kernel = kernel
+        if cpp_kernel is not None:
+            self.cpp_kernel = cpp_kernel
```

Comment:
Is it ever valid to be in here with no kernels at all?

Reply:
Yes, the subclasses do that and define kernel as a class variable.